plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsSchema.php

```diff
@@ -85,6 +86,29 @@ public function get_schema_properties() {
 		];
 	}
 
+	/**
+	 * Validate the field rules.
+	 *
+	 * @param DocumentObject $document_object The document object to validate.
+	 * @param array          $rules The rules to validate against.
+	 * @return bool
+	 */
+	public function validate_document_object_rules( DocumentObject $document_object, $rules ) {
+		$validator = new Validator();
+		$result    = $validator->validate(
+			Helper::toJSON( $document_object->get_data() ),
+			Helper::toJSON(
+				[
+					'$schema'    => 'http://json-schema.org/draft-07/schema#',
+					'type'       => 'object',
+					'properties' => $rules,
```

Comment:
Weird you didn't run into issues of `additionalProps`?

Reply:
`additionalProps` default is true.